### PR TITLE
Staging wifi 1996

### PR DESCRIPTION
**Pull request by shivamcandela**
Removed Ordering, Updated directory structure.
added some readme files
added a standard configuration file for pytest supports both static and quali generated
removed some hard coded stuff

### CODE REVIEW

File: libs/cloudsdk/cloudsdk.py [hunk 1]
```diff
@@ -735,8 +742,8 @@ def upgrade_fw(self, equipment_id=None, force_upgrade=False, force_upload=False)
             try:
                 obj = self.equipment_gateway_client.request_firmware_update(equipment_id=equipment_id,
                                                                             firmware_version_id=firmware_id)
-                print("Request firmware upgrade Success! waiting for 100 sec")
-                time.sleep(100)
+                print("Request firmware upgrade Success! waiting for 300 sec")
```
Comment (jaspreetsachdev) [inline libs/cloudsdk/cloudsdk.py:745]: Future methods should concentrate on removing the sleeps

File: libs/lanforge/lf_tests.py [hunk 2]
```diff
@@ -1,19 +1,20 @@
 #########################################################################################################
 # Used by Nightly_Sanity
-# This has different types of tests like Single client connectivity, Single_Client_EAP, testrail_retest
+# This has different types of old_pytest like Single client connectivity, Single_Client_EAP, testrail_retest
```
Comment (jaspreetsachdev) [inline libs/lanforge/lf_tests.py:3]: old_pytest does not mean anything to the rest of the community

File: tests/configuration.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+
```
Comment (jaspreetsachdev) [inline tests/configuration.py:1]: Header file explaining how to build this would be great.

File: tests/configuration.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
```
Comment (jaspreetsachdev) [inline tests/configuration.py:2]: I worry about user credentials etc to be visible to the rest of the community.

File: tests/controller/test_cloud_basic.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/controller/test_cloud_basic.py:1]: file header documentation. Please be consistent in documenting. We want to the community to follow suit by example.

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+@pytest.mark.bridge
+@pytest.mark.nat
+@pytest.mark.vlan
+@pytest.mark.firmware
+class TestFirmware(object):
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
+        if upload_firmware != 0:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:14]: How does this work when testrail is disabled?

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+@pytest.mark.sanity
+@pytest.mark.bridge
+@pytest.mark.nat
+@pytest.mark.vlan
+@pytest.mark.firmware
+class TestFirmware(object):
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:12]: I would rename instantiate_testrail to something more generic

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:1]: Header and class documentation does not exist. Please add.

File: tests/controller/test_cloud_profiles.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/controller/test_cloud_profiles.py:1]: Header and class level documentation does not exist.

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
+        if upload_firmware != 0:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
+                                                 status_id=1,
+                                                 msg='Create new FW version by API successful')
+            PASS = True
+        else:
+            instantiate_testrail.update_testrail(case_id=test_cases["create_fw"], run_id=instantiate_project,
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:19]: Can't we move the upgrade_testrail to the bottom and pass in value of status and msg from the if/else condition?

File: tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py [hunk 11]
```diff
@@ -0,0 +1,181 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py:1]: Documentation please

File: tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+    'setup_profiles, create_profiles',
+    [(["BRIDGE"], ["BRIDGE"])],
+    indirect=True,
+    scope="class"
+)
+@pytest.mark.usefixtures("setup_profiles")
+@pytest.mark.usefixtures("create_profiles")
+class TestSetupBridge:
+
+    # @pytest.mark.open
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py:17]: Cleanup old and unwanted code.

File: tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py [hunk 13]
```diff
@@ -0,0 +1,181 @@
+import pytest
+
+
+@pytest.mark.sanity
+@pytest.mark.setup_bridge
+@pytest.mark.bridge
+@pytest.mark.parametrize(
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup_e2e_bridge.py:7]: Is there documentation of parametrize behaviour? Maybe a readme on how to add a new test case, new test class and new test module. Then adding a new marker etc.

File: tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py [hunk 14]
```diff
@@ -0,0 +1,181 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py:1]: Documentation

File: tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+    'setup_profiles, create_profiles',
+    [(["VLAN"], ["VLAN"])],
+    indirect=True,
+    scope="class"
+)
+@pytest.mark.usefixtures("setup_profiles")
+@pytest.mark.usefixtures("create_profiles")
+class TestSetupvlan:
+
+    # @pytest.mark.open
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py:17]: Remove code.

File: tests/e2e/basic/client_connectivity_test/test_bridge_mode.py [hunk 16]
```diff
@@ -1,4 +1,7 @@
 import pytest
+
+pytestmark = [pytest.mark.client_connectivity_test, pytest.mark.bridge]
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/client_connectivity_test/test_bridge_mode.py:3]: What does this do? Documentation

File: tests/e2e/basic/conftest.py [hunk 17]
```diff
@@ -0,0 +1,286 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/conftest.py:1]: Documentation

File: tests/e2e/basic/wifi_capacity_test/test_bridge_mode.py [hunk 18]
```diff
@@ -0,0 +1,19 @@
+import pytest
```
Comment (jaspreetsachdev) [inline tests/e2e/basic/wifi_capacity_test/test_bridge_mode.py:1]: Documentation

File: README.md [hunk 19]
```diff
@@ -22,35 +22,51 @@ This project is built using python 3 and strongly recommends using virtualenv to
 
 ## Code style
```
[the remark this answers is not in the document]
Comment (shivamcandela) [reply README.md:23]: Sure, I will clean up the readme and write accordingly and then update here

File: libs/cloudsdk/README.md [hunk 20]
```diff
@@ -35,27 +35,27 @@ Now your cloud sdk code is downloaded with all of the dependencies and you can s
...
+docker build -t wlan-cloud-test -f Dockerfile .
+docker build -t wlan-cloud-lint -f Dockerfile-lint .
 ```
 
 and then you can do something like this to lint your code:
 
 ```shell
-docker run -it --rm -v %path_to_this_dir%/tests wlan-tip-lint -d protected-access *py # for now
-docker run -it --rm -v %path_to_this_dir%/tests wlan-tip-lint *py # for future
+docker run -it --rm -v %path_to_this_dir%/old_pytest wlan-tip-lint -d protected-access *py # for now
```
[the remark this answers is not in the document]
Comment (shivamcandela) [reply libs/cloudsdk/README.md:45]: thats the directory for the pytest structure that resides in current master, its deleted from the repo now, i will remove it from here as well

File: libs/cloudsdk/cloudsdk.py (hunk 1 above)
Comment (shivamcandela) [reply libs/cloudsdk/cloudsdk.py:745]: I am worried if the cloudsdk method itself is self sufficient to verify the firmware upgrade. i wanna explore ucentral for that maybe

File: libs/lanforge/lf_tests.py (hunk 2 above)
Comment (shivamcandela) [reply libs/lanforge/lf_tests.py:3]: sure i will cleanup from here

File: tests/configuration.py (hunk 3 above)
Comment (shivamcandela) [reply tests/configuration.py:1]: i am planning to have a dummy file with dummy data that needs to be configured manually before running testbed,
later on, the class structure will be self explainatory along with some comments, but for now an explaination in comments will be usefull. i will update it

File: tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py (hunk 14 above)
Comment (shivamcandela) [reply tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py:1]: I have started to write readmes, for each testbed environment, that will explain the setup needed for test cases. i will update here once i complete

File: tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py (hunk 15 above)
Comment (shivamcandela) [reply tests/e2e/basic/_basic_test_setup/setup_e2e_vlan.py:17]: that i commented to have to have a template that is required to add more ssid security modes in the configuration part

File: tests/e2e/basic/client_connectivity_test/test_bridge_mode.py (hunk 16 above)
Comment (shivamcandela) [reply tests/e2e/basic/client_connectivity_test/test_bridge_mode.py:3]: this is a module level markering, as a part of pytest framework documentation and best practices, i am mentioning the ways to marker the test cases in an standard formate

File: tests/configuration.py (hunk 4 above)
Comment (shivamcandela) [reply tests/configuration.py:2]: we can write a dummy credentials for now, and keep that configuration file in confluence page somewhere

File: tests/controller/test_cloud_basic.py (hunk 5 above)
Comment (shivamcandela) [reply tests/controller/test_cloud_basic.py:1]: sure

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py (hunk 6 above)
Comment (shivamcandela) [reply tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:14]: so i am having a dummy class that has same methods name, so if we want to disable testrail, then that class instance is passed to all testcase and that function does nothing

File: tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py (hunk 7 above)
Comment (shivamcandela) [reply tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:12]: what are your suggestions?
i am thinking of reporting

Comment (jaspreetsachdev) [reply tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py:12]: Reporting makes sense

**Review by jaspreetsachdev — DISMISSED** [review]

**Review by jaspreetsachdev — DISMISSED** [review]

**Review by jaspreetsachdev — APPROVED** [review]

**Review by shivamcandela — COMMENTED** [review]
Updated readmes

**Pull request merged**